### PR TITLE
Split God-Object DataStore into 3 components

### PR DESCRIPTION
**Pull request by rinslow**
Splitted `DataStore` into:
* `Configurations` class
* `Events` class
* `Measurements` class

### CODE REVIEW

File: data/configurations.py [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
+    Note that this class is implemented with the 'Singleton' design pattern,
+    since access to it is needed all across the application and only a single
+    instance of it would make sense.
+    """
     CONFIG_FILE = os.path.abspath(os.path.join(THIS_DIRECTORY, "..", "config.json"))
-    SYSTEM_SAMPLE_INTERVAL = 22 #KHZ
+    DEFAULT_CONFIG_FILE = os.path.abspath(os.path.join(THIS_DIRECTORY, "..", "default_config.json"))
 
-    MS_TO_SEC = 1000
+    __configurations_instance = None
```
Comment (avrahamshukron) [inline data/configurations.py]: Why not simply 
```suggestion
    __instance = None
```

File: data/configurations.py [hunk 2]
```diff
@@ -31,37 +35,41 @@ def __init__(self, flow_threshold, volume_threshold,
...
+        if cls.__configurations_instance is not None:
+            return cls.__configurations_instance
 
-        self.volume = 0
-        self.flow_measurements = Queue(maxsize=40)
-        self.pressure_measurements = Queue(maxsize=40)
-        self.x_axis = range(0, self.samples_in_graph_amount)
+        instance = cls._load()
+        cls.__configurations_instance = instance
+        return instance
```
Comment (avrahamshukron) [inline data/configurations.py]:
```suggestion
        cls.__configurations_instance = cls._load()
        return cls.__configurations_instance
```

File: data/configurations.py [hunk 3]
```diff
@@ -31,37 +35,41 @@ def __init__(self, flow_threshold, volume_threshold,
...
+    def _load(cls):
+        try:
+            return cls._parse_config_file(cls.CONFIG_FILE)
 
-        self.alerts_queue = AlertsQueue()
-        self.lock = Lock()
+        except ConfigurationFileError:
+            # The second call to _parse_config_file might fail,
+            # we will however let the exception propagate upwards
+            return cls._parse_config_file(cls.DEFAULT_CONFIG_FILE)
```
Comment (avrahamshukron) [inline data/configurations.py]: We should somehow signal that something awful happened.
AT LEAST error log message but maybe we should think about how to notify the user.

This should be done in a different issue - but please put an error log message for now.

File: data/configurations.py [hunk 4]
```diff
@@ -84,15 +92,10 @@ def load_from_config(cls):
...
-                       graph_seconds=12,
-                       breathing_threshold=3.5)
-
-    def save_to_file(self):
+            raise ConfigurationFileError("Could not load config file %s"
+                                         % config_file) from e
+
+    def save_to_file(self, config_path=CONFIG_FILE):
         log.info("Saving threshold values to database")
         config = {
```
Comment (avrahamshukron) [inline data/configurations.py]: There is a huge problem here:
On one side, you have a `default_configuration.json`, and on the other hand you reconstruct the entire JSON when saving.

This is bound to get out of sync. for example a key is added to the default json but not to the save method, and then the load will fail next time.

Why won't we use https://pydantic-docs.helpmanual.io/ or any other library for configuration?

File: main.py [hunk 5]
```diff
@@ -55,7 +59,7 @@ def configure_logging(level, store):
     logger.addHandler(fh)
     logger.addHandler(ch)
     logger.addHandler(sh)
-    logger.disabled = not store.log_enabled
+    logger.disabled = not config.log_enabled
```
Comment (avrahamshukron) [inline main.py]: Why do we need that in the config file? there is no UI to change that and the logging level is controlled from the command line

File: data/events.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import logging
+
+from data.alerts import AlertsQueue
+
+log = logging.getLogger(__name__)
+
+
+class Events(object):
```
[the remark this answers is not in the document]
Comment (rinslow) [reply data/events.py]: As of now it simply abstracts alert queue, in the future we might need more message queues 

preparation for an ac ;P

File: data/configurations.py (hunk 4 above)
Comment (rinslow) [reply data/configurations.py]: put(), get() and full() are atomic
meaning they will run without context switches in the middle

however between calls there is no guarantees for no context switch to happen, we want to atomize the entire transax6

File: data/measurements.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    def __init__(self):
+        self.volume = 0
+        self.flow_measurements = Queue(maxsize=40)  # TODO: Rename?
+        self.pressure_measurements = Queue(maxsize=40) # TODO: Rename?
+        self.x_axis = range(0, self._amount_of_samples_in_graph)
+        self.intake_peak_flow = 0
+        self.intake_peak_pressure = 0
+        self.bpm = 0
+
+        self.lock = Lock()
```
[the remark this answers is not in the document]
Comment (rinslow) [reply data/measurements.py]: put(), get() and full() are atomic
meaning they will run without context switches in the middle

however between calls there is no guarantees for no context switch to happen, we want to atomize the entire transaction

**Pull request merged**